src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -40,7 +40,7 @@ public static bool IsLiteralNull(this BoundExpression node)
 
         public static bool IsLiteralDefault(this BoundExpression node)
         {
-            return node.Kind == BoundKind.DefaultExpression && node.Syntax.Kind() == SyntaxKind.DefaultLiteralExpression;
+            return (node.Kind == BoundKind.DefaultExpression || node.Kind == BoundKind.ConvertedDefaultExpression) && node.Syntax.Kind() == SyntaxKind.DefaultLiteralExpression;
```

Comment:
>node.Syntax.Kind() == SyntaxKind.DefaultLiteralExpression [](start = 118, length = 57)

The syntax check looks suspicious. Why would semantic meaning of a bound node depend on syntax node it is associated with?